### ISSUE TITLE
Leverage browser caching - Not a good first feature

### ISSUE THREAD

**Issue by LukeSchlangen** [approachable-io/approachable-io#5]
This issue is more complicated than most of the issues we recommend starting on since it is more complicated and less approachable than many of the other issues available. If you're looking for your first issue, we recommend starting with one that has a [good first issue](https://github.com/approachable-io/approachable-io/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) label.

From google page speed test: https://developers.google.com/speed/pagespeed/insights/?url=approachable.io&tab=mobile

> Leverage browser caching
Setting an expiry date or a maximum age in the HTTP headers for static resources instructs the browser to load previously downloaded resources from local disk rather than over the network.
Leverage browser caching for the following cacheable resources:
http://www.approachable.io/approachable-open-source-logo.png (10 minutes)
http://www.approachable.io/styles.css (10 minutes)

Suggestions for how to fix can be found here:
https://developers.google.com/speed/docs/insights/LeverageBrowserCaching

**Comment by codyogden** [comment]
Github Pages does not let users define their own browser caching policies (default is to expire in 10 minutes). One suggestion I found is to [use a CDN](https://chrishallahan.com/blog/pass-google-pagespeed-performance-test) like [CloudFlare](https://blog.cloudflare.com/secure-and-fast-github-pages-with-cloudflare/), but it requires DNS-level changes.

A bonus would also be free SSL through CloudFlare which doesn't hurt.

**Comment by LukeSchlangen** [comment]
Thanks for looking into that! Any downside to moving to CloudFlare?  I can't really tell from the guide if it will:

- add complexity making the project less approachable?
- add additional cost to the project?

**Comment by codyogden** [comment]
One downside to using CloudFlare is that it can delay the live site's refresh after merging updates into `master`, but that would happen with the use of any CDN.

**Will it add complexity which makes the project less approachable?**

It should not affect the approachability for anyone to contribute. Nothing in the repo changes, and our existing workflow is unaffected. There exist alternatives like using custom hosting and a deployment process to give us more control in defining caching policies, but those options add both complexity and cost while reducing the approachability.

**Will it add additional cost to the project?**

[CloudFlare has a free tier](https://www.cloudflare.com/plans/) which includes the shared SSL, and it should not create any cost. The free tier comes with three "Page Rules." We'd use at least one to set a custom caching header for our static assets, and we could potentially set longterm expirations for guaranteed assets (like images).

**Comment by LukeSchlangen** [comment]
Ok, I think I have migrated over and everything is cached for 4 hours, but it seems like google is still unhappy with that.

<img width="933" alt="screen shot 2017-10-12 at 9 07 16 am" src="https://user-images.githubusercontent.com/5384581/31500414-f438a82e-af2c-11e7-976c-13b791f67757.png">

Apparently, it only counts as caching to google if it's 8 days (or somewhere between 5 and 8).

<img width="1280" alt="screen shot 2017-10-12 at 9 58 20 am" src="https://user-images.githubusercontent.com/5384581/31502994-180c3264-af34-11e7-8e0c-bc18d9f208d6.png">

After doing all of that work, the google score moved from 90/100 to 91/100! Not a huge change, but I do like that it is now `https` Thanks for all of your help with research @codyogden! 

But... then when I enforced SSL it dropped back to 74/100 because it says we're requiring redirects. I'm probably going to revert for now, but I appreciate all your help!
